tests/test_status_data.py

```diff
@@ -74,19 +75,222 @@ def test_replica_scenario(self):
+
+        time.sleep(0.01)
+
+        self.invalid_pod.error_image_pull()
+        self.status_data.update({"object": self.invalid_pod})
+        assert self.status_data.status_data[release].get("status") == "Image Error"
+
+        # Now there are two pods in the release, one older Running and one newer Image Error
+
+        # Pod: valid_pod
+        self.valid_pod = Pod()
+        self.valid_pod.create(release)
+
+        time.sleep(0.01)
+
+        self.valid_pod.running()
+        self.status_data.update({"object": self.valid_pod})
+        assert self.status_data.status_data[release].get("status") == "Running"
+
+        # The first two pods are deleted but the last pod should remain running
+        self.pod.delete()
+        self.invalid_pod.delete()
+
+        self.status_data.update({"object": self.pod})
+
+        msg = f"Release created ts={self.status_data.status_data[release].get("creation_timestamp")}, \
+                deleted ts={self.status_data.status_data[release].get("deletion_timestamp")}"
+
+        print(msg)
+
+        self.assertEqual(
+            self.status_data.status_data[release].get("status"),
+            "Running",
+            f"Release should be Running after delete of first pod, \
+                         ts pod deleted={self.pod.metadata.deletion_timestamp} vs \
+                         ts invalid_pod deleted={self.invalid_pod.metadata.deletion_timestamp} vs \
+                         ts valid_pod created={self.valid_pod.metadata.creation_timestamp}, {msg}",
+        )
+
+        self.status_data.update({"object": self.invalid_pod})
+        self.assertEqual(
+            self.status_data.status_data[release].get("status"),
+            "Running",
+            f"Release should be Running after delete of 2nd invalid pod, \
+                         ts pod deleted={self.pod.metadata.deletion_timestamp} vs \
+                         ts invalid_pod deleted={self.invalid_pod.metadata.deletion_timestamp} vs \
+                         ts valid_pod created={self.valid_pod.metadata.creation_timestamp}, {msg}",
+        )
+
+
+class TestStatusConverter(unittest.TestCase):
+    """Verifies the translation logic of k8s status objects to app status codes.
+
+    This executes static method determine_status_from_k8s with signature:
+    determine_status_from_k8s(status_object: V1PodStatus) -> Tuple[str, str, str]
+    The response object has structure: status_object: phase, message, pod_message
+    """
+
+    def test_waiting_container_reason_pending(self):
+        """
+        This scenario tests a k8s pod status object with a container with the following status attributes:
+        state=waiting, reason=PodInitializing
+        """
+        podstatus = PodStatus()
+        podstatus.add_container_status("waiting", "PodInitializing")
+        expected = ("Pending", "", "")
+        actual = StatusData.determine_status_from_k8s(podstatus)
+        self.assertEqual(actual, expected)
```

Comment:
This and the tests bellow are perfect case for a parameterized test cases. It would reduce test file size quite a bit I think
But if you want to stick to unittest, then there is a subTest context manager that could be used for this

Reply:
I see your point but the tests are not so similar as they might seem at first. There are different methods for init containers and non-init containers, and one test adds both objects. So parameterized tests could combine at most 2 or 3 tests into one.